src/libraries/System.Collections/ref/System.Collections.cs

```diff
@@ -556,6 +556,25 @@ public abstract partial class EqualityComparer<T> : System.Collections.Generic.I
         bool System.Collections.IEqualityComparer.Equals(object? x, object? y) { throw null; }
         int System.Collections.IEqualityComparer.GetHashCode(object obj) { throw null; }
     }
+
+    public static class EqualityComparer
+    {
+        public static System.Collections.Generic.IEqualityComparer<TEnumerable> CreateEnumerableComparer<TEnumerable, T>(System.Collections.Generic.IEqualityComparer<T>? elementComparer = null) 
+            where TEnumerable : System.Collections.Generic.IEnumerable<T> { throw null; }
+
+        public static System.Collections.Generic.IEqualityComparer<TSet> CreateSetComparer<TSet, T>(System.Collections.Generic.IEqualityComparer<T>? elementComparer = null) 
+            where TSet : System.Collections.Generic.IReadOnlySet<T> { throw null; }
+
+        public static System.Collections.Generic.IEqualityComparer<TDictionary> CreateDictionaryComparer<TDictionary, TKey, TValue>(System.Collections.Generic.IEqualityComparer<TKey>? keyComparer = null, System.Collections.Generic.IEqualityComparer<TValue>? valueComparer = null) 
+            where TDictionary : System.Collections.Generic.IReadOnlyDictionary<TKey, TValue> { throw null; }
+    }
+
+    public static class ComparerFactory
```

Comment:
The approved API uses the `Comparer` class in the `System.Collections.Generic` namespace, but it is [already present](https://learn.microsoft.com/en-us/dotnet/api/system.collections.comparer) in the `System.Collections` namespace
https://github.com/dotnet/runtime/blob/5b8ebeabb32f7f4118d0cc8b8db28705b62469ee/src/libraries/System.Runtime/ref/System.Runtime.cs#L7251-L7258

Issue reported [here](https://github.com/dotnet/runtime/issues/77209#issuecomment-1356557303)